src/component_constructors/thermalgeneration_constructor.jl

```diff
@@ -1,100 +1,134 @@
-###Dispatch Formulations##
-
 """
-This function creates the minimal themal dispatch formulation depending on combination of devices, device_formulation and system_formulation
+This function creates the model for a full themal dispatch formulation depending on combination of devices, device_formulation and system_formulation
 """
-function constructdevice!(m::JuMP.AbstractModel, netinjection::BalanceNamedTuple, category::Type{PowerSystems.ThermalGen}, category_formulation::Type{D}, system_formulation::Type{S}, sys::PowerSystems.PowerSystem; args...) where {D <: AbstractThermalDispatchForm, S <: PM.AbstractPowerFormulation}
+function constructdevice!(ps_m::CanonicalModel, category::Type{T}, category_formulation::Type{D}, system_formulation::Type{S}, sys::PSY.PowerSystem; kwargs...) where {T <: PSY.ThermalGen, D <: AbstractThermalFormulation, S <: PM.AbstractPowerFormulation}
 
-    p_th = activepowervariables(m, sys.generators.thermal, sys.time_periods);
+    #Defining this outside in order to enable time slicing later
+    time_range = 1:sys.time_periods
 
-    varnetinjectiterate!(netinjection.var_active, p_th, sys.time_periods, sys.generators.thermal)
+    #Variables
+    activepowervariables(ps_m, sys.generators.thermal, time_range);
 
-    activepower(m, sys.generators.thermal, category_formulation, system_formulation, sys.time_periods)
+    reactivepowervariables(ps_m, sys.generators.thermal, time_range);
 
-    cost = variablecost(m, sys.generators.thermal, category_formulation, system_formulation)
+    commitmentvariables(ps_m, sys.generators.thermal, time_range)
 
-    add_to_cost!(m, cost)
+    #Constraints
+    activepower(ps_m, sys.generators.thermal, category_formulation, system_formulation, time_range)
 
-end
+    reactivepower(ps_m, sys.generators.thermal, category_formulation, system_formulation, time_range)
 
+    if :initial_conditions in keys(kwargs)
 
-function constructdevice!(m::JuMP.AbstractModel, netinjection::BalanceNamedTuple, category::Type{PowerSystems.ThermalGen}, category_formulation::Type{D}, system_formulation::Type{S}, sys::PowerSystems.PowerSystem; args...) where {D <: AbstractThermalDispatchForm, S <: AbstractACPowerModel}
+        initial_conditions = kwargs[:initial_conditions]
 
-    constructdevice!(m, netinjection, category, category_formulation, PM.AbstractPowerFormulation, sys; args...)
+    else
 
-    q_th = reactivepowervariables(m, sys.generators.thermal, sys.time_periods);
+        initial_conditions = [(g.name, Float64(g.tech.activepower > 0.0)) for g in sys.generators.thermal]
```

Comment:
Should this be `Int64` instead?

Reply:
I haven't implemented the initial conditions passing yet.